app/dash/Accounts/Add/AddKeystore/index.js

```diff
@@ -1,251 +1,111 @@
-                      onChange={(e) => this.onChange('keystorePassword', e)}
-                      onFocus={(e) => this.onFocus('keystorePassword', e)}
-                      onBlur={(e) => this.onBlur('keystorePassword', e)}
-                      onKeyPress={(e) => this.keyPress(e, () => this.next())}
-                    />
-                  </div>
-                  <div className='addAccountItemOptionSubmit' onMouseDown={() => this.next()}>
-                    Next
-                  </div>
-                </div>
-                <div className='addAccountItemOptionSetupFrame'>
-                  <div className='addAccountItemOptionTitle'>Create Account Password</div>
-                  <div className='addAccountItemOptionInput addAccountItemOptionInputPassword'>
-                    <div className='addAccountItemOptionSubtitle'>
-                      password must be 12 characters or longer
-                    </div>
-                    <input
-                      type='password'
-                      tabIndex='-1'
-                      ref={this.forms.keystoreCreatePassword}
-                      value={this.state.password}
-                      onChange={(e) => this.onChange('password', e)}
-                      onFocus={(e) => this.onFocus('password', e)}
-                      onBlur={(e) => this.onBlur('password', e)}
-                      onKeyPress={(e) => this.keyPress(e, () => this.createKeystore())}
-                    />
-                  </div>
-                  <div className='addAccountItemOptionSubmit' onMouseDown={() => this.createKeystore()}>
-                    Create
-                  </div>
-                </div>
-                <div className='addAccountItemOptionSetupFrame'>
-                  <div className='addAccountItemOptionTitle'>{this.state.status}</div>
-                  {this.state.error ? (
-                    <div className='addAccountItemOptionSubmit' onMouseDown={() => this.restart()}>
-                      try again
-                    </div>
-                  ) : null}
-                </div>
-              </div>
-            </div>
-          </div>
-          <div className='addAccountItemFooter' />
-        </div>
-      </div>
-    )
-  }
+  const steps = [
+    <LocateKeystore key={0} {...{ addKeystore, error, setError }} />,
+    <Loading key={1} />,
+    <EnterKeystorePassword key={2} keystore={accountData.keystore} />
+  ]
+  return <>{steps[viewIndex]}</>
 }
 
-export default Restore.connect(AddRing)
+const AddKeystore = ({ accountData }) => (
+  <AddHotAccount
+    {...{
+      title: 'Key Store',
+      summary: 'A keystore account lets you add accounts from your keystore.json file',
+      svgName: 'file',
+      intro: 'Add KeyStore Account',
+      accountData,
+      createSignerMethod: 'createFromKeystore',
+      firstStep: <LoadKeystore key={0} accountData={accountData} />,
+      newAccountType: 'keystore',
+      validateSecret: () => {}
+    }}
+  />
```

Comment:
```suggestion
  <AddHotAccount
      title='Key Store',
      summary='A keystore account lets you add accounts from your keystore.json file'
      svgName='file'
      intro='Add KeyStore Account'
      accountData={accountData}
      createSignerMethod='createFromKeystore',
      firstStep={<LoadKeystore key={0} accountData={accountData} />}
      newAccountType='keystore'
      validateSecret={() => {}}
    }}
  />
```

Reply:
That won't commit correctly from GH because it's only pointing at the last line but you get the idea